adapters/repos/db/shard.go

```diff
@@ -433,7 +431,7 @@ func (s *Shard) initVectorIndex(ctx context.Context,
 		return nil, fmt.Errorf("Unknown vector index type: %q. Choose one from [\"%s\", \"%s\"]",
 			vectorIndexUserConfig.IndexType(), vectorindex.VectorIndexTypeHNSW, vectorindex.VectorIndexTypeFLAT)
 	}
-
+	defer vectorIndex.PostStartup()
```

Comment:
already mentioned by @asdine in the backport for 1.23

`nit: no need to defer here`